src/checker.ts

```diff
@@ -149,13 +106,18 @@ export function generateReport(
   }
 
   // Go through and check all the assertions.
-  for (const {assertion, type, error} of nodedAssertions) {
+  for (const noded of nodedAssertions) {
+    let { assertion, type, error, node } = noded;
+    let { pos, end } = node;
+    let code = source.text.substring(pos, end);
```

Comment:
Is this the same as `node.getText()`?

If not, I always prefer to use string.slice. See http://www.danvk.org/2014/11/17/javascript-string-splice-substr-substring-which-to-use.html